swarm/storage/mru/request.go

```diff
@@ -230,6 +228,27 @@ func (j *updateRequestJSON) decode() (*Request, error) {
 	return r, nil
 }
 
+func decodeHexArray(dst []byte, src string, expectedLength int, name string) error {
```

Comment:
These are internal helper functions. The idea is that they check and return an error with a message depending on the field name that we are decoding. Before, I was checking outside, and it was repeated code all over.

Reply:
Ok, but the part which enforces a specific length is still generic, no?

Reply:
To your first comment:
>decodeHexArray should either return a byte slice, or the length should be inferred from the byte slice provided. The former of which seems to me to be more in line with go style.

I was able to refactor `decodeHexArray` to infer the length out of the slice. The other one I think that would add us an extra copy since internally this calls `hexutil.Decode()` which returns a new slice, so I would then have to copy it to ours...

>Ok, but the part which enforces a specific length is still generic, no?

Yes, I guess it could be reused, but is it worth it?

Reply:
Likewise you could argue it worth having `hexutil.Decode` and `hexutil.MustDecode` for exampe. Let's put it in the aforementioned discussion about directly populating passed slices.